### ISSUE TITLE
Optional mutation argument passed to where changes entire table

### ISSUE THREAD

**Issue by TeoTN** [hasura/graphql-engine#4859]
I've noticed that when an optional mutation argument is passed to `where` clause, it changes the entire table:

* query:
```
mutation Attach(
  $nextParentId: uuid,
  $nextParentChildren: jsonb!
) {
  attach: update_model(
  where: {id: {_eq: $nextParentId}}, _set: {children: $nextParentChildren}) {
    returning {
      id
      children
    }
  }
}
```
* variables:
```
{ children: "[]" }
```

That query has emptied all `children` arrays across all entities in the table.

**Comment by tirumaraiselvan** [comment]
This might be the reason: https://github.com/hasura/graphql-engine/issues/704

**Comment by tirumaraiselvan** [comment]
This is now fixed in v2.0 where explicit `null` values will give an error.
